catroid/src/main/java/org/catrobat/catroid/utils/CrashReporter.java

```diff
@@ -0,0 +1,148 @@
+	public static final String EXCEPTION_FOR_REPORT = "EXCEPTION_FOR_REPORT";
+
+	static Context context;
+	private static SharedPreferences preferences;
+	private static boolean isCrashReportEnabled;
+	private static boolean isCrashlyticsInitialized;
+
+	private CrashReporter() {
+	}
+
+	public static boolean initialize(Context context) {
+
+		CrashReporter.context = context;
+		preferences = PreferenceManager.getDefaultSharedPreferences(context);
+		setIsCrashReportEnabled(BuildConfig.CRASHLYTICS_CRASH_REPORT_ENABLED);
+		isCrashlyticsInitialized = false;
+
+		if (isAnonymousReportEnabled() && isCrashReportEnabled) {
+			Fabric.with(context, new Crashlytics());
+			Log.d(TAG, "INITIALIZED!");
+			isCrashlyticsInitialized = true;
+			return true;
+		}
+		Log.d(TAG, "INITIALIZATION FAILED! [ Anonymous Report : " + isAnonymousReportEnabled() + " Crash Report : " + isCrashReportEnabled + " ]");
+		isCrashlyticsInitialized = false;
+		return false;
+	}
+
+	private static boolean isAnonymousReportEnabled() {
+		return preferences != null && preferences.getBoolean(SettingsActivity.SETTINGS_CRASH_REPORTS, false);
+	}
+
+	private static boolean isRecoveredFromCrash() {
+		return preferences != null && preferences.getBoolean(RECOVERED_FROM_CRASH, false);
+	}
+
+	public static void disableCrashlytics() {
+		if (isCrashlyticsInitialized) {
+			Crashlytics crashlytics = new Crashlytics.Builder()
+					.core(new CrashlyticsCore.Builder().disabled(BuildConfig.DEBUG).build())
+					.build();
+			Fabric.with(context, crashlytics);
+			isCrashlyticsInitialized = false;
+		}
+	}
+
+	@VisibleForTesting
+	public static void setIsCrashReportEnabled(boolean value) {
+		isCrashReportEnabled = value;
+	}
+
+	public static boolean logException(Throwable t) {
+
+		if (isAnonymousReportEnabled() && isCrashReportEnabled) {
+			Log.e(TAG, "LOG_EXCEPTION : " + t);
+			Crashlytics.logException(t);
+			return true;
+		}
+
+		return false;
+	}
+
+	public static boolean checkIfCrashRecoveryAndFinishActivity(final Activity context) {
+		Log.d(TAG, "AFTER_EXCEPTION : checkIfCrashRecoveryAndFinishActivity()");
+		if (isRecoveredFromCrash()) {
+			if (isCrashReportEnabled && isAnonymousReportEnabled()) {
+				sendUnhandledCaughtException();
+			}
+			if (!(context instanceof MainMenuActivity)) {
+				context.finish();
+			} else {
+				preferences.edit().putBoolean(RECOVERED_FROM_CRASH, false).commit();
+				return true;
+			}
+		}
+		return false;
+	}
+
+	public static void sendUnhandledCaughtException() {
+		Log.d(TAG, "AFTER_EXCEPTION : sendCaughtException()");
+		Gson gson = new Gson();
+		String json = preferences.getString(EXCEPTION_FOR_REPORT, "");
+		Throwable exception = gson.fromJson(json, Throwable.class);
+		logException(exception);
+		preferences.edit().remove(EXCEPTION_FOR_REPORT).commit();
+	}
+
+	public static void storeUnhandledException(Throwable exception) {
+		SharedPreferences.Editor prefsEditor = preferences.edit();
+		if (isCrashReportEnabled) {
```

Comment:
This check (`isCrashReportEnabled && isAnonymousReportEnabled()`) is used often. Please create a method for it.